backend/src/api/node_context.py

```diff
@@ -146,3 +146,11 @@ def storage_dir(self) -> Path:
 
         This directory persists between node executions, and its contents are shared between different nodes.
         """
+
+    @abstractmethod
+    def add_cleanup(self, fn: Callable) -> None:
+        """
+        Registers a function that will be called when the chain execution is finished.
+
+        The function will be called with no arguments.
+        """
```

Comment:
Why document what can be expressed with types?

Also, we should add the guarantee about registering the same function to the docs.

```suggestion
    def add_cleanup(self, fn: Callable[[], None]) -> None:
        """
        Registers a function that will be called when the chain execution is finished.
        
        Registering the same function (object) twice will only result in the function being called once.
        """
```